pkg/objstore/s3/s3.go

```diff
@@ -250,7 +250,14 @@ func (b *Bucket) Exists(ctx context.Context, name string) (bool, error) {
 
 // Upload the contents of the reader as an object into the bucket.
 func (b *Bucket) Upload(ctx context.Context, name string, r io.Reader) error {
-	_, err := b.client.PutObjectWithContext(ctx, b.name, name, r, -1,
+	// TODO(PR #617): Consider removing requirement on pre-known length when all providers will support this.
+	fileInfo, err := r.(*os.File).Stat()
+	if err != nil {
+		return errors.Wrapf(err, "could not stat local file %s", name)
```

Comment:
Well, we agreed for best effort logic, so we need warning instead I think, right?

Reply:
My bad. I really need holidays :face_with_head_bandage: